mitodl/odl-video-service
# Decouple watch bucket uploads from collection titles

#### What are the relevant tickets?
Fixes #271 

#### What's this PR do?
- Creates a new `slug` field on `Collection`, and the migration script pre-populates it with the current collection title for all non-TechTV collections.
- Modifies the watch-bucket monitoring/upload task to use `slug` instead of `title` for determining which collection the videos should be uploaded to.

#### How should this be manually tested?
- Set `VIDEO_S3_WATCH_BUCKET=odl-video-service-uploaded-ci` in your `.env` file
- Set `VIDEO_WATCH_BUCKET_FREQUENCY=30` in your `.env` file
- `docker-compose run web python manage.py migrate`
- Copy some small video files over into your `odl-video-service` folder
- In a django shell:
   ```python
    from django.conf import settings
    from ui.utils import get_bucket
    bucket = get_bucket(settings.VIDEO_S3_WATCH_BUCKET)
    bucket.upload_file('<your_video_1>', 'MIT-8.03-2018-Spring-lec-mit-0000-2018may01-1333-L01.mp4')
    bucket.upload_file('<your_video_2>', 'MIT-8.03-2018-Spring-lec-mit-0000-2018may01-1333-L02.mp4')
    ```
- Within 30 seconds the watch-bucket monitor task should run and create two new collections, with `title` and `slug` == `'MIT-8.03-2018-Spring-L01` and `'MIT-8.03-2018-Spring-L02'`, and each containing one video.
- Change the titles to something different, like 'MIT 8.03 Spring Semester' or 'Best Course Ever'. The slugs should not be changed.
- Upload some more videos to the bucket:
    ```python
    bucket.upload_file('<your_video_3>', 'MIT-8.03-2018-Spring-lec-mit-0000-2018may02-2666-L01.mp4')
    bucket.upload_file('<your_video_4>', 'MIT-8.03-2018-Spring-lec-mit-0000-2018may02-2666-L02.mp4')
    ```
- Within 30 seconds the watch-bucket monitor task should run and place the above videos into the same two collections that already exist, it should not create any new collections.
- Manually create a new collection and add a video to it.  Nothing should go wrong.  If you check the new collection in the django admin, the `slug` field should be empty.
